### PR TITLE
Make some code more efficient

### PR DESCRIPTION
**Pull request by liuminghui233**
In this PR, 
1. Reduce the use of `Stream`.
2. Reduce the use of `String.format()`.
3. Add cached object in `DatasetHeader` to reduce repeated calculations.
4. Simplify the constructor of `FunctionExpression`.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/ExpressionAnalyzer.java [hunk 1]
```diff
@@ -1107,33 +1107,10 @@ public static boolean checkIsNeedTransform(Expression expression) {
...
-      return getDeviceNameInSourceExpression(((BinaryExpression) expression).getRightExpression());
-    } else if (expression instanceof UnaryExpression) {
-      return getDeviceNameInSourceExpression(((UnaryExpression) expression).getExpression());
-    } else if (expression instanceof TimeSeriesOperand) {
+    if (expression instanceof TimeSeriesOperand) {
       return ((TimeSeriesOperand) expression).getPath().getDevice();
     } else if (expression instanceof FunctionExpression) {
-      return getDeviceNameInSourceExpression(expression.getExpressions().get(0));
-    } else if (expression instanceof ConstantOperand || expression instanceof TimestampOperand) {
-      return null;
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/ExpressionAnalyzer.java:1136]: why delete this branch?

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/ExpressionAnalyzer.java [hunk 2]
```diff
@@ -1107,33 +1107,10 @@ public static boolean checkIsNeedTransform(Expression expression) {
...
+      return ((TimeSeriesOperand) expression.getExpressions().get(0)).getPath().getDevice();
```
Comment (JackieTien97) [inline server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/ExpressionAnalyzer.java:1113]: why don't need to call `getDeviceNameInSourceExpression` recursively.

**Review by JackieTien97 — CHANGES_REQUESTED** [review]

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/ExpressionAnalyzer.java (hunk 1 above)
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/ExpressionAnalyzer.java:1136]: The method's argument must be a time series operand.

File: server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/ExpressionAnalyzer.java (hunk 2 above)
Comment (liuminghui233) [reply server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/ExpressionAnalyzer.java]: The method's argument must be a time series operand.

**Review by JackieTien97 — APPROVED** [review]

**Pull request merged**